assets/src/block-validation/use-validation-error-state-updates.js

```diff
@@ -76,31 +77,118 @@ export function maybeAddClientIdToValidationError( { validationError, source, cu
  * components within multiple slotfills to have access to the same state.
  */
 export function useValidationErrorStateUpdates() {
-	const [ trackedValidationErrorsFromPost, setTrackedValidationErrorsFromPost ] = useState( [] );
-
-	const { setValidationErrors } = useDispatch( BLOCK_VALIDATION_STORE_KEY );
-
-	const { blockOrder, currentPost, getBlock, validationErrorsFromPost } = useSelect( ( select ) => ( {
-		blockOrder: select( 'core/block-editor' ).getClientIdsWithDescendants(),
+	const [ blockOrderBeforeSave, setBlockOrderBeforeSave ] = useState( [] );
+	const [ hasRequestedPreview, setHasRequestedPreview ] = useState( false );
+	const [ previousValidationErrors, setPreviousValidationErrors ] = useState( [] );
+	const [ shouldValidate, setShouldValidate ] = useState( true );
+	const unmounted = useRef( false );
+
+	const { setIsFetchingErrors, setReviewLink, setValidationErrors } = useDispatch( BLOCK_VALIDATION_STORE_KEY );
+
+	const {
+		currentPost,
+		getBlock,
+		getClientIdsWithDescendants,
+		isAutosavingPost,
+		isPreviewingPost,
+		isSavingPost,
+		previewLink,
+		validationErrors,
+	} = useSelect( ( select ) => ( {
+		getClientIdsWithDescendants: select( 'core/block-editor' ).getClientIdsWithDescendants,
 		currentPost: select( 'core/editor' ).getCurrentPost(),
 		getBlock: select( 'core/block-editor' ).getBlock,
-		validationErrorsFromPost: select( 'core/editor' ).getEditedPostAttribute( AMP_VALIDITY_REST_FIELD_NAME )?.results || [],
+		getBlocks: select( 'core/block-editor' ).getBlocks(),
+		isAutosavingPost: select( 'core/editor' ).isAutosavingPost(),
+		isPreviewingPost: select( 'core/editor' ).isPreviewingPost(),
+		isSavingPost: select( 'core/editor' ).isSavingPost(),
+		previewLink: select( 'core/editor' ).getEditedPostPreviewLink(),
+		validationErrors: select( BLOCK_VALIDATION_STORE_KEY ).getValidationErrors(),
 	} ), [] );
 
-	/*
+	// Set unmounted to true on unmount to prevent state updates after async functions.
+	useEffect( () => () => {
+		unmounted.current = true;
+	}, [] );
+
+	/**
+	 * Set flags when a post is being saved. Validation should not be triggered on autosaves
+	 * with an exception of an autosave initiated by a post preview request.
```

Comment:
It should also be triggered when clicking the “Re-validate now” button in the sidebar.

Reply:
This has been addressed in da8e35a.